tests/github-action-helper.sh

```diff
@@ -104,6 +104,37 @@ wait_for_operator_pod_to_be_ready_state_custom() {
 EOF
 }
 
+wait_for_pod_running() {
+  DEPLOYMENT="$1"
+  NAMESPACE="$2"
+  TIMEOUT="10"  # in seconds
+
+  BASH_CMD="${BASH_CMD:=bash}" # might be 'bash -x', for example
+
+  msg="Waiting ${TIMEOUT} seconds for pod from deployment ${DEPLOYMENT} to be running..."
+  echo "${msg} countdown: ${TIMEOUT}"
+  start_time=$SECONDS
+  debug_pod
+  if [[ $DEPLOYMENT == "rook-ceph-osd-0" ]]
+  then
+    debug_pod=$(kubectl get pods -n ${NAMESPACE} | grep ${DEPLOYMENT} | awk '!/rook-ceph-osd-0-debug/' | awk '{ print $1  }')
```

Comment:
How can this be changed so the pod name isn't hard coded in the helper?

Reply:
@travisn So a special case because,
The debug-osd-0 pod still runs in the cluster, and if we go and fetch the osd-0 pod from the deployment name,
It returns the:
debug-osd-0 pod and osd-0-pod name, 
So to distinguish between them we need to have this special condition.

Another thing we can do is to watch for debug-osd-0 pod to be deleted, But I am not sure if we can do it for the CI as we don't know the debug+pod name, and if go and fetch the name using the deployment name the above problem mentioned needs to handle.

There is one thing that @BlaineEXE mentioned we can do,
For checking that the debug pod is deleted successfully before scaling back the original deployment in the actual code,
BUt this would add an additional wait to the application which is not required.

Reply:
Actually, I think we can replace this function with a `kubectl wait` condition. In #61 I have a helper to [wait_for_pod_of_deployment_to_be_running](https://github.com/rook/kubectl-rook-ceph/pull/61/files#diff-8545dbb7aa4b5f9f70a1fc5bfb63a6ea985db01f32b89ec751eda93824d03651R247-R251). I believe it's the same thing you're trying to accomplish with this wait. Can you try that instead?

Reply:
I thought the `rook-ceph-osd` pod is even not required so removed it

Reply:
Did you look if we could use the wait_for_pod_of_deployment_to_be_running in this case as well? This function seems like it has the same purpose.

Reply:
Yaa we can use that function too for this case.

Reply:
But this current function provide more details,
SHould we go ahead with that as it looks simpler in terms of code...

Reply:
@travisn they end up doing the same task,

But your function looks more robust becasue it directly checks the deployment resource without fetching the pod name and checking,
So updated to it.